pkg/emulator/emulator/cartesi-machine-emulator.go

```diff
@@ -0,0 +1,967 @@
+package emulator
+
+/*
+#cgo LDFLAGS: -lcartesi -lcartesi_jsonrpc
+#include "machine-c-api.h"
+#include "jsonrpc-machine-c-api.h"
+#include <stdlib.h>
+*/
+import "C"
+
+import (
+	"encoding/hex"
+	"unsafe"
+)
+
+type Error int32
```

Comment:
Go convention: its best to return the `error` type.

Functions currently are: `func foo() (something *Something, errMsg string, errCode Error)`.
It would be best if they were: `func foo() (something *Something, err error)`.

Then, we could have a special struct type:

```
type Error struct {
	Code int32
	Msg string
}
```

That implements the `error` interface:

```
func (err Error) String() string {
	return "cartesi-machine error: " + err.Code.String() + " (" + err.Msg + ")"
}
```

The function that instantiates the `Error` struct and returns an `error` could be something like this:

```
func newError(code C.int, msg *C.char) error {
	defer C.cm_delete_cstring(msg)
	if code := mapCCodeToGoCode(C.CM_ERROR(code)); code != ErrorOk {
		return errors.New(Error{Code: code, Msg: C.GoString(message)}.String())
	}
	return nil
}
```